setup.py

```diff
@@ -48,26 +49,32 @@
 
 def get_git_revision():
     """Try to obtain the current git revision number"""
-    xia2_root_path = os.path.split(os.path.realpath(__file__))[0]
+    xia2_root_path = Path(__file__).resolve().parent
 
-    if not os.path.exists(os.path.join(xia2_root_path, ".git")):
+    if not xia2_root_path.joinpath(".git").is_dir():
         return None
 
     try:
         result = subprocess.run(
-            ("git", "describe", "--long"),
+            ("git", "describe", "--tags", "--long", "--first-parent"),
             check=True,
-            cwd=xia2_root_path,
```

Comment:
I believe this should not be removed?

Reply:
Hmm, apparently pip explicitly moves the CWD to the installing package. At least we're more robust to other running with it explicitly set.